### ISSUE TITLE
Last file gets lost

### ISSUE THREAD

**Issue by iislucas** [klaascuvelier/gulp-copy#3]
See: https://github.com/ivogabe/gulp-typescript/issues/75#issuecomment-77334504

To reproduce: 
1. clone this repo: 
https://github.com/uProxy/node-chrome-runner/tree/f56d0b0b5eaf6e15947c9f953649706e28b4d770
2. Run gulp and observe there is a type-error in chrome-runner.ts
3. Remove the file src/foo.ts (note it is empty), so that only a single file is compiled
4. Run gulp and observe: no errors are printed, no JS file is created. grunt-typescript ignored the file. :(

Feedback from Gulp devs was: 

> It looks like this is an issue with gulp-copy. The last file passed to gulp-copy is removed from the 
> stream. In step 2 foo.ts is not passed through, only chrome-runner.ts is passed through to gulp-
> typescript. In step 4 the last file, chrome-runner.ts is removed from the stream, thus no file is passed to 
> gulp-typescript. Can you log an issue here?

so here is the issue :)

**Comment by klaascuvelier** [comment]
Thanks for reporting, I'll take a look at this.

**Comment by klaascuvelier** [comment]
@iislucas how does the tsd stuff work? It appears I am missing files if I just clone the repo and checkout the given commit. 

````
➜  node-chrome-runner git:(f56d0b0) ✗ gulp
[gulp] Using gulpfile ~/Projects/node-chrome-runner/gulpfile.js
[gulp] Starting 'typescript-src'...
build/dev/node-chrome-runner/chrome-runner.ts(1,1): 6053 File 'build/third_party/typings/node/node.d.ts' not found.
build/dev/node-chrome-runner/chrome-runner.ts(18,18): 1005 ',' expected.
build/dev/node-chrome-runner/chrome-runner.ts(19,16): 1005 ',' expected.
build/dev/node-chrome-runner/chrome-runner.ts(22,1): 1005 ',' expected.
````

**Comment by iislucas** [comment]
Sorry, I should have said, you just need to run `./setup.sh install` first and it will run the relevant tsd command, npm install, etc for you.

**Comment by klaascuvelier** [comment]
@iislucas I keep getting this:
````
➜  node-chrome-runner git:(f56d0b0) ✗ gulp
[16:24:52] Using gulpfile ~/Projects/node-chrome-runner/gulpfile.js
[16:24:52] Starting 'typescript-src'...
build/dev/node-chrome-runner/chrome-runner.ts(18,18): 1005 ',' expected.
build/dev/node-chrome-runner/chrome-runner.ts(19,16): 1005 ',' expected.
build/dev/node-chrome-runner/chrome-runner.ts(22,1): 1005 ',' expected.
[16:24:52] Finished 'typescript-src' after 260 ms
[16:24:52] Starting 'default'...
[16:24:52] Finished 'default' after 4.91 μs
````

**Comment by iislucas** [comment]
oh, that's odd, I'll have a look. Sorry, I should have seen that from your previous comment that it wasn't related to install... I'll check...

**Comment by iislucas** [comment]
Looking again at this example, the errors are exactly what is supposed to happen! :)  The bug demo code seems to still do the same thing for me. 

The problem is that if you remove the `foo.ts` file from src, you don't get them. Gulp copy has a bug when there is a single file in the directory. Although the file is copied to the correct output location, the gulp stream of files is empty and so the compilation attempt doesn't get made. 

Make sense?

**Comment by DanielJoyce** [comment]
I might be bumping into this as well. Or something similar. Copy task is not pulling js files generated by typescript into my dist location.

**Comment by klaascuvelier** [comment]
This is issue should be fixed in `1.0.0`
